model/project_ref_test.go

```diff
@@ -266,6 +268,52 @@ func TestGetActivationTimeWithCron(t *testing.T) {
 	}
 }
 
+func TestChangeOwnerRepo(t *testing.T) {
+	require.NoError(t, db.ClearCollections(ProjectRefCollection, RepoRefCollection, evergreen.ScopeCollection,
+		evergreen.RoleCollection, user.Collection, evergreen.ConfigCollection))
+	env := evergreen.GetEnvironment()
+	_ = env.DB().RunCommand(nil, map[string]string{"create": evergreen.ScopeCollection})
+	settings := testutil.TestConfig()
+	settings.GithubOrgs = []string{"evergreen-ci"}
+	settings.GithubOrgs = []string{"newOwner"}
+	assert.NoError(t, evergreen.UpdateConfig(settings))
+
+	evergreen.SetEnvironment(env)
+	pRef := ProjectRef{
+		Id:              "myProject",
+		Owner:           "evergreen-ci",
+		Repo:            "evergreen",
+		Admins:          []string{"me"},
+		RepoRefId:       "myRepo",
+		UseRepoSettings: true,
+	}
+	assert.NoError(t, pRef.Insert())
+	repoRef := RepoRef{ProjectRef{
+		Id: "myRepo",
+	}}
+	assert.NoError(t, repoRef.Upsert())
+	u := &user.DBUser{Id: "me",
+		SystemRoles: []string{GetViewRepoRole("myRepo")},
+	}
+	assert.NoError(t, u.Insert())
+	pRef.Owner = "newOwner"
+	pRef.Repo = "newRepo"
+	assert.NoError(t, pRef.ChangeOwnerRepo(u))
```

Comment:
Would it maybe makes sense to change the flow a bit to do this part in the function? Like have the function take in the new owner and repo, and set it in the function, as opposed to having whoever calls this(most likely a graphql resolver) set it first on the projectRef and then calling ChangeOwnerRepo? 

(this is a comment on the overall setup, not the test specifically).

Reply:
I thought about that -- I'm not sure, it really depends on how information is passed to the resolver I guess. The other Edit functions assume that the ProjectRef itself has been modified and then just updates the relevant pieces so I thought this would be more consistent. (I think this also covers the above question.) If you think it's better the other way though I can change it to take in owner/repo.

Reply:
Either way works, and I hear the argument for consistency. I'll approve and leave it up to you to decide.

Reply:
I'll leave it and it can be a quick change if it turns out easier for the UI to have it work the other way 👍